### PR TITLE
added external wallet-providers to massa-web3

### PR DESCRIPTION
**Pull request by BatiGencho**

### CODE REVIEW

File: src/web3/Client.ts [hunk 1]
```diff
@@ -82,6 +94,74 @@ export class Client implements IClient {
...
+      this.externalWalletProvidersDiscoveryTimeoutId.clear();
+  }
+
+  /**
+   * A method to start listening for a connection to Thyra's server.
+   *
+   * @returns void
+   */
+  public startClientDiscovery(): void {
+    const that = this;
```
Comment (peterjah) [inline src/web3/Client.ts:157]: this should not be needed if you use "this" in arrow function. which is the case line 163

Comment (BatiGencho) [reply src/web3/Client.ts]: Great catch, thanks!

Comment (BatiGencho) [reply src/web3/Client.ts]: Fixed

File: src/web3/Client.ts [hunk 2]
```diff
@@ -82,6 +94,73 @@ export class Client implements IClient {
...
+    this.startClientDiscovery = this.startClientDiscovery.bind(this);
+
+    // start client discovery
+    if (withExternalProviderDiscovery) {
+      this.startClientDiscovery();
+    }
+  }
+
+  /**
+   * A callback method that triggers a ping of the Thyra's server
```
[the remark this answers is not in the document]
Comment (BatiGencho) [reply src/web3/Client.ts]: Right, it was a bad copy paste from the wallet-provider. I have fixed that now, thanks for the remarsk.

**Review by Thykof — APPROVED** [review]

**Review by gregLibert — APPROVED** [review]

**Pull request merged**